### ISSUE TITLE
Dinsar_gacos_error

### ISSUE THREAD

**Issue by Xin-Wang-520** [esrivas17/gacos_correction_for_GMTSAR#9]
Hello, Professor. According to your manual, single_GACOS_correction.csh can be used for atmospheric correction of dinsar. I put the files of two dates of gacos into the merge folder generated after p2p_S1_TOPS_Frame. single_GACOS_correction.csh 20230209.ztd 20230209.ztd.rsc 20230221.ztd 20230221.ztd.rsc phasefilt_interp.grd ref_point 33.8, and then I get an error. Is that right? If yes, this error is reported. How to solve it?

![2023-03-10 11-09-35 的屏幕截图](https://user-images.githubusercontent.com/96721536/224213523-82c8a92d-b520-48e9-aaeb-3c31272c79c4.png)

**Comment by Xin-Wang-520** [comment]
@esrivas17 @erik-17

**Comment by Xin-Wang-520** [comment]
@esrivas17 @erik-17 Professor, if you could solve the problem above

**Comment by erik-17** [comment]
Hi Sorry for the late response. Not sure what could go wrong. Seems like it didnt even pass the first steps. I wonder what are the dimensions of your GACOS grids and also of your interferograms.

**Comment by erik-17** [comment]
if you could share the log of `gmt grdinfo` of your interfogram and gacos

**Comment by Xin-Wang-520** [comment]
> The error suggests that maybe your interferogram is larger than the GACOS grids?

phase_GACOS_corrected_phs_detrended.grd I have solved, have learned the atmosphere correction of matlab, later try your csh script, I mainly interference grd write wrong, thank you. Professor, I would like to ask you three questions. The first one is reference_point, is it set randomly? I have interviewed in matlab before. And the second incidence_angle, where are you looking? Are you looking in the sentry files? And the third phase_GACOS_corrected_phs.grd and phase_GACOS_corrected_phs_detrended. The final atmospheric correction we want is phase_GACOS_corrected_phs_detrended.grd this?

**Comment by esrivas17** [comment]
The reference point in theory must be a location where you have the certain that is a stable area, since in many cases we don't have access to that information is tended to be chosen a point with a high coherence value.
in GMTSAR there is a script called SAT_look, if I remember correctly, you can take the average of the incidence angles.
If you detrend the correction or not will depend on the nature of your deformation, for cases like land subsidence is common to detrend it, and cases like earthquakes where the deformation is big it is not recommended to detrend since you could filter out part of the deformation signal.
Hope that helps.
Additionally I would like to recommend to try the correction over the unwrapped phase, at the moment I am not sure if the correction should be applied to the wrapped or unwrapped phase. You could experiment with this and take the best result.

**Comment by Xin-Wang-520** [comment]
> a high coherence value

Thank you,Professor. 1. The choice of reference point, what you mean by a high coherence value, is a point far away from the deformation field with no or small deformation, or a point with a deformation field similar to the epicenter? I think most people take the point away from the deformation field. 2. Can you tell me in detail how to use it? I only used it to take enu direction vector before, for example, SAT_look master.PRM < topo.llt > topo.lltn

**Comment by esrivas17** [comment]
Yes you are right, I meant that also with an stable area, so preferably that is not within the main deformation area. But also make sure is in a high coherence area. 
one of the outputs of SAT_look is the elevation, the incidence angle  if I am not mistaken is the complement of the elevation value.

**Comment by ditafaith** [comment]
Hi @esrivas17 @erik-17 @Xin-Wang-520 , sorry to ask this question. I already tried and find the suitable case but can't fully capture the problem.  Is there any problem with input files?
For reference,
------------
(base) faith@book merge % gmt grdinfo phasefilt_ll.grd 
phasefilt_ll.grd: Title: 
phasefilt_ll.grd: Command: gmt xyz2grd llpb -R-241.3625/-238.845833333/-1.94166666667/0.0958333333333 -I1.5s/1.5s -r -fg -Gphasefilt_ll.grd -bi3f
phasefilt_ll.grd: Remark: 
phasefilt_ll.grd: Pixel node registration used [Geographic grid]
phasefilt_ll.grd: Grid file format: nf = GMT netCDF format (32-bit float), CF-1.7
phasefilt_ll.grd: x_min: -241.3625 x_max: -238.845833333 x_inc: 0.000416666666722 name: longitude n_columns: 6040
phasefilt_ll.grd: y_min: -1.94166666667 y_max: 0.0958333333335 y_inc: 0.000416666666667 name: latitude n_rows: 4890
phasefilt_ll.grd: v_min: -3.14158892632 v_max: 3.14158678055 name: z
phasefilt_ll.grd: scale_factor: 1 add_offset: 0
phasefilt_ll.grd: format: netCDF-4 chunk_size: 129,129 shuffle: on deflation_level: 3
-----------
(base) faith@book merge % more 20260613.ztd.rsc 
WIDTH   4626
FILE_LENGTH   4254
X_FIRST          118.2862240
Y_FIRST            0.1049190
X_STEP             0.000833333330000
Y_STEP            -0.000833333330000
X_UNIT        degres
Y_UNIT        degres
Z_OFFSET      0
Z_SCALE       1
PROJECTION    LATLON
DATUM         WGS84
-------------

<img width="1191" height="342" alt="Image" src="https://github.com/user-attachments/assets/d78af224-3011-4c4a-a46c-aa172f64dca8" />
